### PR TITLE
Escape regular expressions

### PR DESCRIPTION
**Pull request by jcblw**
## Description

This fixes some issues with trying to replace and parse text. The regular expressions would fail if we passed in text like "This (text)", and this allows that to be properly used.

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ * @param {string} text The string to escape.
+ * @returns {string} Returns the escaped string.
+ */
+
+export const escapeRegExp = (text: string) => {
+  return text && reHasRegExpChar.test(text)
+    ? text.replace(reRegExpChar, "\\$&")
+    : text || "";
+};
```
Comment (ShaunMerritt) [inline src/utils.ts:16]: Oh this is super helpful!

File: src/index.ts [hunk 2]
```diff
@@ -1,3 +1,7 @@
+import { escapeRegExp } from "./utils";
+
+export { escapeRegExp } from "./utils";
```
Comment (mikeyamato) [inline src/index.ts:3]: why export from here when you're already doing it from within `utils.ts`?

File: src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ * "?", "(", ")", "[", "]", "{", "}", and "|" in `string`.
+ *
+ * @param {string} text The string to escape.
+ * @returns {string} Returns the escaped string.
+ */
+
+export const escapeRegExp = (text: string) => {
+  return text && reHasRegExpChar.test(text)
+    ? text.replace(reRegExpChar, "\\$&")
+    : text || "";
```
Comment (mikeyamato) [inline src/utils.ts:15]: if there's nothing to escape it returns the give text, right?

**Review by mikeyamato — APPROVED** [review]

File: src/index.ts (hunk 2 above)
Comment (jcblw) [reply src/index.ts:3]: To use it in a project you would need to do `import { escapeRegExp } from 'aracari/dist/utils';` to use it in utils. With this export you can use `import { escapeRegExp } from 'aracari';`

File: src/utils.ts (hunk 3 above)
Comment (jcblw) [reply src/utils.ts:15]: yep

**Pull request merged**